code/modules/research/machinery/pointgiver.dm

```diff
@@ -0,0 +1,16 @@
+/obj/machinery/rnd/pointgiver
```

Comment:
This could as well just be a /obj/structure subtype in the current state.
But I aknowledge it's pretty WIP at the moment and perhaps you may have some design in mind that justifies them being a machinery/rnd subtype.
Also they'll need an icon_state.